jina/peapods/pod.py

```diff
@@ -82,15 +80,27 @@ def _parse_args(self, args: Namespace) -> Dict[str, Optional[Union[List[Namespac
             'tail': None,
             'peas': []
         }
-
         if getattr(args, 'parallel', 1) > 1:
             # reasons to separate head and tail from peas is that they
             # can be deducted based on the previous and next pods
             peas_args['head'] = _copy_to_head_args(args, args.polling.is_push)
             peas_args['tail'] = _copy_to_tail_args(args)
             peas_args['peas'] = _set_peas_args(args, peas_args['head'], peas_args['tail'])
+            self.is_head_router = True
+            self.is_tail_router = True
+        elif getattr(args, 'uses_before', None) or getattr(args, 'uses_after', None):
+            args.scheduling = SchedulerType.ROUND_ROBIN
+            if getattr(args, 'uses_before', None):
+                peas_args['head'] = _copy_to_head_args(args, args.polling.is_push)
+                self.is_head_router = True
+            if getattr(args, 'uses_after', None):
+                peas_args['tail'] = _copy_to_tail_args(args)
+                self.is_tail_router = True
+            peas_args['peas'] = _set_peas_args(args, peas_args.get('head', None), peas_args.get('tail', None))
         else:
             peas_args['peas'] = [args]
+            self.is_head_router = False
```

Comment:
Moving these `self.is_head_router = False` and `self.is_tail_router = False` to the top of the function makes it clearer the default behavior and clarifies @maximilianwerk comment on wehter u can have a `head` without a `tail` and viceversa

Reply:
__init__ already has it